src/Storages/StorageMaterializedView.cpp

```diff
@@ -153,6 +153,7 @@ StorageMaterializedView::StorageMaterializedView(
 
     if (query.refresh_strategy)
     {
+        fixed_uuid = query.refresh_strategy->append;
```

Comment:
Some more changes needed for that PR? Can't see `append` in 
https://github.com/ClickHouse/ClickHouse/blob/48bea29611b078d2db2cb8f2e10ff0be56f6099d/src/Parsers/ASTRefreshStrategy.h#L18-L22


Why described issue with race-condition is related only to append?